transformation/plugins/org.jboss.tools.fuse.transformation.editor/src/main/java/org/jboss/tools/fuse/transformation/editor/internal/util/Util.java

```diff
@@ -111,14 +109,19 @@ public static boolean draggingFromValidTarget(TransformationManager manager) {
     }
 
     public static void ensureSourceFolderExists(IJavaProject javaProject,
-                                                String folder,
-                                                IProgressMonitor monitor) throws JavaModelException {
+                                                String folderName,
+                                                IProgressMonitor monitor) throws Exception {
         boolean exists = false;
         IClasspathEntry[] entries = javaProject.getRawClasspath();
-        IPath path = javaProject.getPath().append(folder);
+        IPath path = javaProject.getPath().append(folderName);
         for (IClasspathEntry entry : entries) {
             if (entry.getEntryKind() == IClasspathEntry.CPE_SOURCE && entry.getPath().equals(path)) {
                 exists = true;
+                File folder = javaProject.getResource().getLocation().append(folderName).toFile();
+                if (!folder.exists()) {
+                    if (!folder.mkdirs()) throw new Exception(Messages.bind(Messages.Util_UnableToCreateSourceFolder, folder));
```

Comment:
I would normally agree regarding a more specific exception, but in this case we already know up front that if the source folder cannot be created or determined for _any_ reason, the wizard cannot continue, regardless of whether the cause is due to a system I/O problem or something else.  The only important distinction will ever be the error message.  IOW, I'm confident there will be no later on in this case.

Regarding the style issue, I'm not sure how the traditional approach is any more scientific.  I'm arguing that the single line is actually more readable and avoids ambiguity.  I believe the single line makes the statement visually flow better as read and makes it very clear the if statement has no additional content (else it would appear in a block).  There's a reason the language allows for statements to be made outside of blocks.